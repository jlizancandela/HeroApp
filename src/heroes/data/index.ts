import { Hero } from "./types";
import data from "./heroes.json";
export const heroes: Hero[] = data;
export * from "./types";
