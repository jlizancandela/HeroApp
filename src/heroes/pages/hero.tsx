import { Navigate, useNavigate, useParams } from "react-router-dom";
import Aos from "aos";
import "aos/dist/aos.css";
import { useEffect } from "react";
import { getHero } from "../helper/gethero";

type Params = {
  id: string;
};

export const HeroPage = () => {
  useEffect(() => {
    Aos.init();
  }, []);

  const { id } = useParams<Params>();

  const hero = id ? getHero(id) : null;
  const navigate = useNavigate();

  if (!hero) {
    return <Navigate to="/" replace />;
  }

  const onBack = () => {
    navigate(-1);
  };

  return (
    <article className="d-flex flex-row">
      <div className="p-3" data-aos="fade-right" data-aos-duration="2000">
        {hero?.id && (
          <img src={`/heroes/${hero?.id}.jpg`} alt={hero?.superhero} />
        )}
      </div>
      <div className="d-flex flex-column p-3">
        <h1>{hero?.superhero}</h1>
        <hr />
        <p>{hero?.alter_ego}</p>
        <p>{hero?.characters !== hero?.alter_ego && hero?.characters}</p>
        <p>{hero?.first_appearance}</p>
        <p>{hero?.publisher}</p>
        <button
          type="button"
          className="btn btn-outline-primary"
          onClick={onBack}
        >
          Back
        </button>
      </div>
    </article>
  );
};
