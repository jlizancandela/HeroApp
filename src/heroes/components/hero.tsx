import { useEffect } from 'react';
import AOS from 'aos';
import { Hero } from '../data/types';

import 'aos/dist/aos.css';

export const HeroCard = ({ hero }: { hero: Hero }) => {
  useEffect(() => {
    AOS.init();
  }, []);

  return (
    <div className="card" style={{ width: '18rem' }} data-aos="fade-up" data-aos-duration="2000">
      <img src={`/heroes/${hero.id}.jpg`} className="card-img-top" alt={hero.id} />
      <div className="card-body d-flex flex-column flex-fill">
        <h5 className="card-title">{hero.superhero}</h5>
        <p className="card-text">{hero.alter_ego}</p>
        <p className="card-text">{hero.characters !== hero.alter_ego && hero.characters}</p>
        <a className="btn btn-dark mt-auto" href={`/hero/${hero.id}`}>
          Mas informaciÃ³n
        </a>
      </div>
    </div>
  );
};
