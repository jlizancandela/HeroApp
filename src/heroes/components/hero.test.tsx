import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { HeroCard } from './hero';
import { Hero } from '../data';

describe('HeroCard', () => {
  let hero: Hero;

  beforeEach(() => {
    hero = {
      id: '1',
      alter_ego: 'test alter ego',
      characters: 'test characters',
      first_appearance: 'test first appearance',
      publisher: 'test publisher',
      superhero: 'test superhero',
    };
  });

  afterEach(() => {
    hero = {} as Hero;
  });

  it('should render correctly', () => {
    const { container } = render(<HeroCard hero={hero} />);
    expect(container).toMatchSnapshot();
    expect(screen.getByText(hero.alter_ego)).toBeInTheDocument();
    expect(screen.getByText(hero.characters)).toBeInTheDocument();
    expect(screen.getByText(hero.superhero)).toBeInTheDocument();
  });
});
