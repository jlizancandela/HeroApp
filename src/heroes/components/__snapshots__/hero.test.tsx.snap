// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeroCard > should render correctly 1`] = `
<div>
  <div
    class="card aos-init aos-animate"
    data-aos="fade-up"
    data-aos-duration="2000"
    style="width: 18rem;"
  >
    <img
      alt="1"
      class="card-img-top"
      src="/heroes/1.jpg"
    />
    <div
      class="card-body d-flex flex-column flex-fill"
    >
      <h5
        class="card-title"
      >
        test superhero
      </h5>
      <p
        class="card-text"
      >
        test alter ego
      </p>
      <p
        class="card-text"
      >
        test characters
      </p>
      <a
        class="btn btn-dark mt-auto"
        href="/hero/1"
      >
        Mas informaciÃ³n
      </a>
    </div>
  </div>
</div>
`;
